src/main/java/uk/gov/cslearning/record/domain/ModuleRecord.java

```diff
@@ -4,21 +4,26 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
 import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.hibernate.annotations.NotFound;
 import org.hibernate.annotations.NotFoundAction;
+import uk.gov.cslearning.record.domain.converter.BookingStatusConverter;
 
-import javax.persistence.*;
-import javax.validation.constraints.NotBlank;
-import javax.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 
 import static com.google.gson.internal.$Gson$Preconditions.checkArgument;
 
 @Entity
+@Getter
+@Setter
```

Comment:
Could we use `@Data` here?

Reply:
`@Data` brings a bit more than just getters/setters and doesn't always play nicely with `@entiy`. In this case it started causing an infinite recursion stack overflow exception related to the course record relationship because it includes some builtin toString logic...if I remember correctly